File: backend/test/com/skewwhiffy/auraltester/clefs/ClefSpaceNotesTest.scala
package com.skewwhiffy.auraltester.clefs

import com.skewwhiffy.auraltester.scales.Key
import com.skewwhiffy.auraltester.testutils.TestData
import org.scalatest.flatspec.AnyFlatSpec

class ClefSpaceNotesTest extends AnyFlatSpec {
  it should "correctly define line note from clef" in {
    val clef = TestData.noteFactories.clef.treble
    val expectedAbc = List("F", "A", "c", "e")
    val expectedNames = expectedAbc.map(it => it.toUpperCase)

    val actual = ClefSpaceNotes(clef)

    assert(actual.notes.map(it => it.abc(Key.cMajor)) == expectedAbc)
    assert(actual.notes.map(it => it.lyric) == expectedNames.map(it => Some(it)))
  }

}

File: backend.kotlin/src/test/kotlin/com/skewwhiffy/notation/model/abc/ClefSpaceNotesTest.kt
package com.skewwhiffy.notation.model.abc

import com.skewwhiffy.notation.model.key.Key
import com.skewwhiffy.test.util.TestData
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ClefSpaceNotesTest {
  @Test
  fun `correctly define line note from clef`() {
    val clef = TestData.noteFactories.clef.treble
    val expectedAbc = listOf("F", "A", "c", "e")
    val expectedNames = expectedAbc.map { it.uppercase() }

    val actual = ClefSpaceNotes(clef)

    assertThat(actual.notes.map { it.abc(Key.cMajor) }).isEqualTo(expectedAbc)
    assertThat(actual.notes.map { it.lyric }).isEqualTo(expectedNames)
  }
}